### ISSUE TITLE
crash after updating to 1.3.x

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#8128]
## Description 
Multiple reports of crashing after updating to 1.3.x, unable to keep Brave open:
https://twitter.com/brave/status/1225842308014170112
https://twitter.com/brave/status/1225842071900020741
https://twitter.com/brave/status/1225841791569580034

## Steps to Reproduce
Unsure, possibly related to sync

## Actual result:
see below tweet for video of issue
https://twitter.com/bpemble/status/1225838160162115586

## Expected result:
No crash

## Reproduces how often: 

easily for affected users, not all users affected

## Brave version (brave://version info)

1.1.113

## Version/Channel Information:

- Can you reproduce this issue with the current release? yes
- Can you reproduce this issue with the beta channel? unsure
- Can you reproduce this issue with the dev channel? unsure
- Can you reproduce this issue with the nightly channel? unsure

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @brave/legacy_qa @jsecretan @darkdh

**Comment by CharlieRanalli** [comment]
Crash report attached if it helps, sync enabled, same behavior as other affected users on Twitter.

Suggested script produced this:
cr-mac-03:~ charlieranalli$ /Applications/Brave\ Browser.app/Contents/MacOS/Brave\ Browser --disable-brave-sync
[0207/142113.418995:WARNING:process_memory_mac.cc(93)] mach_vm_read(0x7ffeef8fb000, 0x2000): (os/kern) invalid address (1)
[0207/142113.604945:WARNING:system_snapshot_mac.cc(42)] sysctlbyname kern.nx: No such file or directory (2)
[1335:775:0207/142114.269484:ERROR:child_thread_impl.cc(222)] Invalid PlatformChannel receive right
Segmentation fault: 11

[69C290D3-D551-EF20-0168-01D6DDC43F66.txt](https://github.com/brave/brave-browser/files/4172696/69C290D3-D551-EF20-0168-01D6DDC43F66.txt)

**Comment by kevinp2** [comment]
Same issue here.  I have now become dependent upon Brave, and this happened right after the mandatory upgrade.  Please help!

**Comment by LaurenWags** [comment]
@kevinp2 the team is actively working on a fix. However, some users have reported that if Brave stays open for a minute or two, they are able to do the following to stop the crashing:
1. Navigate to brave://sync
2. Disable bookmark sync from this page

Unfortunately if Brave crashes immediately this won't work for you. The team is working on a fix.

**Comment by kevinp2** [comment]
Hi, an update:  I reinstalled Brave from brave.com  Brave now opens properly and I have my bookmarks back.  Phew!

However, I lost my two open windows and tabs which ironically I had just spent time rearranging and organizing.  Is there any way to get them back?

**Comment by jonathansampson** [comment]
@kevinp2 Do you now see sponsored new tab pages (every fourth new tab opened)?

**Comment by kevinp2** [comment]
> @kevinp2 Do you now see sponsored new tab pages (every fourth new tab opened)?

Hi @jonathansampson , not sure what you mean.  I have my history back, but not the two windows and about 20 tabs I had open between them.

**Comment by jonathansampson** [comment]
@kevinp2 Apologies for the confusion. This version of Brave shipped a New Tab Page update, which allows users to earn BAT from the imagery shown on every 4th new tab. If you open a few new tabs, do you eventually see a custom background for _Western Digital_?

**Comment by CharlieRanalli** [comment]
Reinstalling also worked for me - likely because it reverted to 1.2.43

**Comment by jonathansampson** [comment]
@CharlieRanalli Great point. We did revert to 1.2.43 on brave.com.

**Comment by kevinp2** [comment]
> @kevinp2 Apologies for the confusion. This version of Brave shipped a New Tab Page update, which allows users to earn BAT from the imagery shown on every 4th new tab. If you open a few new tabs, do you eventually see a custom background for _Western Digital_?

No, I opened a few tabs by right-clicking but did not see any tab with an ad.

**Comment by CharlieRanalli** [comment]
> @CharlieRanalli Great point. We did revert to 1.2.43 on brave.com.

@jonathansampson Custom backgrounds should therefore be gone.  I didn't get one after serially opening 6-7 new tabs but I wasn't sure if it was because i'm currently out of the country - I don't get ads here either

**Comment by jonathansampson** [comment]
@CharlieRanalli If you're on 1.2.43 (previous working build), you definitely won't be seeing the updated new tab pages. Regarding the lost windows, it sounds as though reverting to 1.2.43 may have overwritten some of the browser state. Unfortunately, I don't think those will be recoverable.

**Comment by smadgerano** [comment]
So, for those of us who don't want to revert back to a previous version because it means losing the current session (30+ tabs), my browsing history (which I use on a daily basis), and all my extension data that I have no idea how to transfer and isn't based in the cloud, what do we do? I deliberately chose the stable channel specifically so stuff like this didn't happen. The whole world would be up in arms if this was any of the other major browsers.

**Comment by jonathansampson** [comment]
We have a build now that should address this issue; please install and let us know if this resolves the issue for you: 

* Win x64 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup.exe
* Win x86 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup32.exe
* macOS - https://github.com/brave/brave-browser/releases/download/v1.3.114/Brave-Browser.dmg
* Linux - https://github.com/brave/brave-browser/releases/download/v1.3.114/brave-browser_1.3.114_amd64.deb

**Comment by mjgiarlo** [comment]
> We have a build now that should address this issue:
> 
> * Win x64 - [`v1.3.114` BraveBrowserStandaloneSetup.exe (download)](https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup.exe)
> * Win x86 - [`v1.3.114` BraveBrowserStandaloneSetup32.exe (download)](https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup32.exe)
> * macOS - [`v1.3.114` Brave-Browser.dmg (download)](https://github.com/brave/brave-browser/releases/download/v1.3.114/Brave-Browser.dmg)
> * Linux - [`v1.3.114` brave-browser_1.3.114_amd64.deb (download)](https://github.com/brave/brave-browser/releases/download/v1.3.114/brave-browser_1.3.114_amd64.deb)

Tested on Ubuntu 18.04 and worked like a charm. Thanks much, @jonathansampson & crew!

**Comment by louistiti** [comment]
> We have a build now that should address this issue:
> 
> * Win x64 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup.exe
> * Win x86 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup32.exe
> * macOS - https://github.com/brave/brave-browser/releases/download/v1.3.114/Brave-Browser.dmg
> * Linux - https://github.com/brave/brave-browser/releases/download/v1.3.114/brave-browser_1.3.114_amd64.deb

Tested on macOS Catalina and it's working now. Thanks to the team!

**Comment by martinib77** [comment]
> 
> 
> We have a build now that should address this issue:
> 
>     * Win x64 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup.exe
> 
>     * Win x86 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup32.exe
> 
>     * macOS - https://github.com/brave/brave-browser/releases/download/v1.3.114/Brave-Browser.dmg
> 
>     * Linux - https://github.com/brave/brave-browser/releases/download/v1.3.114/brave-browser_1.3.114_amd64.deb

Tested on Windows 10 Enterprise N 2016 LTSB 64 bits.
It's Working. Thanks!

**Comment by aemxn** [comment]
I was having this issue today. No chance to get any report on the crash nor disabling sync in the settings as it happened fast. Installing v1.3.114 (64bit) link above works for me.

**Comment by cagrimertbakirci** [comment]
Same problem here. Currently testing turning off sync to see if it helps any. Will report results in a bit. It crashes every 4-5 minutes of use in my case.

**Comment by smadgerano** [comment]
So will installing the new version wipe out any local settings, local extension caches, and local wallet balance?

**Comment by robincsamuel** [comment]
I was facing the same issue, was crashing in a second. I tried the new build, but crashes in a few minutes. 

I see this in the OSX console.

`fault	12:23:29.924971+0530	Brave Browser Helper (GPU)	Failed to get Metal shader cache main folder: failed to lookup DARWIN_USER_CACHE_DIR err:5
`

Version 1.3.114 Chromium: 80.0.3987.87 (Official Build) (64-bit)
OSX version: 10.15.2 (19C57)

> We have a build now that should address this issue:
> 
> * Win x64 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup.exe
> * Win x86 - https://github.com/brave/brave-browser/releases/download/v1.3.114/BraveBrowserStandaloneSetup32.exe
> * macOS - https://github.com/brave/brave-browser/releases/download/v1.3.114/Brave-Browser.dmg
> * Linux - https://github.com/brave/brave-browser/releases/download/v1.3.114/brave-browser_1.3.114_amd64.deb

**Comment by jonathansampson** [comment]
Thank you, @robincsamuel. Could you go to `about:crashes` and submit the last couple of crashes, then restart Brave, and revisit `about:crashes` to collect the new Crash IDs. Pleas share those here when you have a moment.

**Comment by robincsamuel** [comment]
@jonathansampson Submitted the last three crash reports and restarted Brave, it says uploaded.
`4b3481be5291f190`, `9ef5a973878b82d2`, and `86d0aeac2b804b76` are the crash report ids. I've disabled sync and Brave didn't crash after that. I'll update here if it crashes again.

**Comment by lemosley01** [comment]
I've upgraded to the Windows X64 build above on WIn10 and still see the crashes when I enable sync. 

Crash Id 474d1cbc5cbe0785  and I have another pending.

This started this morning after rebooting. I upgraded Nvidia drivers last night and it was stable then.

**Comment by victorelec14** [comment]
maybe is this the problem ?

Utility: Bat Ledger Service , 90-100%  for 2-5 min them crash.

![image](https://user-images.githubusercontent.com/543450/74110121-10f0ad80-4b8a-11ea-81ae-01bc53355de8.png)

**Comment by jonathansampson** [comment]
@victorelec14, happy to look into this. I don't think the ledger service should be consuming that much processing power, but there may be something there to investigate. Can you go to about:crashes and submit a few of your recent ones. Restarting Brave will cause them to submit instantly. Then you'll find Crash IDs on that page. Please share at your convenience.

**Comment by victorelec14** [comment]
@jonathansampson  , here are the last 10 crashes :) thx !

```
Crashes (27)
Uploaded Crash Report ID 1be23ddcb660b2af (Local Context: a72f28d5-adb3-41b7-b90d-f8717b0b07eb)
Crash report captured on Sunday, February 9, 2020 at 11:58:24 AM, uploaded on Sunday, February 9, 2020 at 11:04:46 PM

Uploaded Crash Report ID 11a11ac320923137 (Local Context: 1dd1feff-1618-4150-be6c-a3437f328516)
Crash report captured on Sunday, February 9, 2020 at 10:10:11 AM, uploaded on Sunday, February 9, 2020 at 11:04:45 PM

Uploaded Crash Report ID e9d04bf609c98cf1 (Local Context: 143ee0bd-90ba-4ecf-bf09-3e0462e69715)
Crash report captured on Saturday, February 8, 2020 at 2:07:28 PM, uploaded on Sunday, February 9, 2020 at 11:04:44 PM

Uploaded Crash Report ID 300340f55d3de614 (Local Context: b86f97aa-edfd-4af7-bc3e-c78a7c02fd35)
Crash report captured on Saturday, February 8, 2020 at 2:02:55 PM, uploaded on Sunday, February 9, 2020 at 11:04:44 PM

Uploaded Crash Report ID c25d72df77c69d5d (Local Context: a477bd91-1b52-441e-a203-4146f0dc3d3c)
Crash report captured on Saturday, February 8, 2020 at 1:41:56 PM, uploaded on Sunday, February 9, 2020 at 11:04:43 PM

Uploaded Crash Report ID 6cd4ffddb3308907 (Local Context: 32efbae4-f5a5-4657-aac5-d6c684382fc6)
Crash report captured on Saturday, February 8, 2020 at 12:48:57 PM, uploaded on Sunday, February 9, 2020 at 11:04:42 PM

Uploaded Crash Report ID ebc5b041992a2001 (Local Context: 44513093-b904-4ffd-a0c8-f3c0538726a8)
Crash report captured on Saturday, February 8, 2020 at 12:36:15 PM, uploaded on Sunday, February 9, 2020 at 11:04:41 PM

Uploaded Crash Report ID 2ec35df153c1e1cd (Local Context: 916f6fa2-577f-4ea6-bdf6-3de6481d332f)
Crash report captured on Saturday, February 8, 2020 at 11:58:03 AM, uploaded on Sunday, February 9, 2020 at 11:04:41 PM

Uploaded Crash Report ID e4fe0c4545df690e (Local Context: b3df18f4-8c22-4d4f-9692-1f4573f79c66)
Crash report captured on Saturday, February 8, 2020 at 11:50:19 AM, uploaded on Sunday, February 9, 2020 at 11:04:40 PM

Uploaded Crash Report ID b8a9e40f0fe4fb55 (Local Context: bb121f6d-c04d-49b8-828f-f788f5f7f61f)
Crash report captured on Saturday, February 8, 2020 at 11:44:49 AM, uploaded on Sunday, February 9, 2020 at 11:04:39 PM
```

**Comment by lemosley01** [comment]
FYI, I re-enabled sync and it is not crashing with the new build - nothing has changed hardware or software-wise since I reported it.

**Comment by robincsamuel** [comment]
> FYI, I re-enabled sync and it is not crashing with the new build - nothing has changed hardware or software-wise since I reported it.

I'm observing the same now. To summarize, Yesterday morning, for some reason, Brave wasn't staying open for even a second. I found this thread and downloaded the versions mentioned in the above comment. After that, Brave started crashing every few minutes. I turned off sync and the problem went away. Then I uploaded the crash reports and turned on sync to reproduce the issue but no crashes after that. :)

**Comment by dvmason** [comment]
@jonathansampson Brave offered upgrade on MacOS Mojave... I agreed... installed and crashed. Tried Nightly... same result... tried moving Library/Application Support/BraveSoftware/Brave-Browser out of the way... same result... tried 1.3.114... same result. Not happy... was sepending on Brave.

**Comment by robincsamuel** [comment]
I'm observing crashes again every few minutes. Uploaded crash reports, `e26feb194539c74f`, `bd02cd5bc33c2c47`

**Comment by tmancey** [comment]
@robincsamuel @dvmason Can you please submit a crash report and let me know the crash report id. Thanks

**Comment by robincsamuel** [comment]
@tmancey I've upgraded to 1.3.115. I will upload and share the crash report id here if the issue persists. Thanks!

**Comment by robincsamuel** [comment]
I just noticed that `brave://sync` is not available on 1.3.115, and therefore not sure if sync is on. The issue wasn't there when I turn off sync.

**Comment by dvmason** [comment]
@tmancey I downloaded the latest (1.3.115) and it works fine. Thank you!!!!

As far as I know, I've never uses sync, so that wasn't the problem.

**Comment by bracklanna** [comment]
> I just noticed that `brave://sync` is not available on 1.3.115, and therefore not sure if sync is on. The issue wasn't there when I turn off sync.

Same here, brave://sync is turned off in 1.3.115. I do use BraveSync for bookmarks between Desktop & 2 iOS devices, and have noticed the lack of sync.

**Comment by MrHaroldA** [comment]
> Same here, brave://sync is turned off in 1.3.115.
You are 2 versions behind: in 1.3.118, sync reappeared as a flag.

**Comment by StFS** [comment]
I'm using 1.5.115 on Mac and I can't find the sync option anywhere. Still, this issue is closed. Is there some other issue that is tracking the progress of whatever is going on with this? Being able to sync bookmarks is a pretty important feature for me and I'd like to avoid installing some plugin for doing that.

**Comment by MrHaroldA** [comment]
Did you enable the sync flag, as described above?

chrome://flags

**Comment by StFS** [comment]
@MrHaroldA darn it... missed that explanation.. will try, thanks for the pointer

**Comment by kjozwiak** [comment]
@StFS we added a `brave://flag` as mentioned by @MrHaroldA via https://github.com/brave/brave-browser/issues/8187 in https://github.com/brave/brave-browser/releases/tag/v1.3.118.

**Comment by akash07k** [comment]
Hi all,
If I enable sync via flags in latest Brave Browser Version 81.1.7.98 
@kjozwiak Will I face any crashes/issues? is it safe to enable?

**Comment by saschpe** [comment]
The way this issue was handled really looks amateur to me. First of all, the sync feature was disabled for 100% of the user base, while a crash might have affected a small minority. Then, [all](https://community.brave.com/t/notice-for-users-on-brave-version-1-3-113/106809) [community](https://community.brave.com/t/sync-greyed-out/106899) issues are closed and don't take further comments. One has to dig up this Github issue three months later just to realise that `brave://flag` doesn't exist. Also, do you expect all your users to go to Github to read that random comment and mess with browser internals?

I do like Brave but this isn't the first time that issues are handled in a rather weird way...

**Comment by jsecretan** [comment]
Sorry for the confusion @saschpe, I realize that our messaging on all of this was hard to follow.  We've put together a support article for getting sync v1 back on while we work on v2 https://support.brave.com/hc/en-us/articles/360042159452

**Comment by saschpe** [comment]
Thanks, that is much appreciated. Keep up the good work!

**Comment by akash07k** [comment]
Hi,
Is there any ETA by when we can expect sync v2? I've enabled older
sync but it is crashing my browser

**Comment by bsclifton** [comment]
cc: @jsecretan

**Comment by intel352** [comment]
Is resolution of this issue stalled out now?
